CRM/Upgrade/Incremental/php/FiveFiftyFour.php

```diff
@@ -50,6 +52,17 @@ public static function addCreatedIDColumnToParticipant($ctx): bool {
     return TRUE;
   }
 
+  /**
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function alterDedupeRuleGroupName(CRM_Queue_TaskContext $ctx) {
+    CRM_Core_DAO::executeQuery("ALTER TABLE `civicrm_dedupe_rule_group` CHANGE COLUMN `name` `name` varchar(255) COMMENT 'Unique name of rule group'", [], TRUE, NULL, FALSE, FALSE);
+    CRM_Core_DAO::executeQuery("UPDATE `civicrm_dedupe_rule_group` g1, `civicrm_dedupe_rule_group` g2 SET g1.name = CONCAT(g1.name, '_', g1.id) WHERE g1.name = g2.name AND g1.id > g2.id", [], TRUE, NULL, FALSE, FALSE);
```

Comment:
I think you need to reverse the order of these two statements

Reply:
@eileenmcnaughton why? The first one just makes the column longer, which makes room for the suffix appended in the 2nd statement.

Reply:
Also I think the id > will be safe in practice - but we don't want to rename anything that `is_reserved` as the name has meaning then

Reply:
Actually no - the issue is you aren't creating the indexes - CRM_Core_BAO_SchemaHandler::createIndexes('civicrm_dedupe_rule_group', ['name']);

Reply:
hmm - maybe you are but not here...